### ISSUE TITLE
struct vs class mismatch in 10.1 C++ system library

### ISSUE THREAD

**Issue by ilg-ul** [riscv/riscv-gcc#246]
When I compile C++ code with `-Wmismatched-tags` I get an warning caused by some mismatches in system headers:

```
[64/71] Building CXX object CMakeFiles/micro-os-plus-utils-lists-static.d...ojects.git/blinky-rtos/xpacks/micro-os-plus-utils-lists/src/lists.cpp.obj
In file included from /home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/bits/ios_base.h:46,
                 from /home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/streambuf:41,
                 from /home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/bits/streambuf_iterator.h:35,
                 from /home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/iterator:66,
                 from /home/ilg/Work/xpack-study-projects.git/blinky-rtos/xpacks/micro-os-plus-utils-lists/include/micro-os-plus/utils/lists.h:40,
                 from /home/ilg/Work/xpack-study-projects.git/blinky-rtos/xpacks/micro-os-plus-utils-lists/src/lists.cpp:30:
/home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/system_error:54:9: note: replace the class-key with 'struct'
   54 |   class error_code;
      |         ^~~~~~~~~~
/home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/system_error:180:10: note: 'std::error_code' defined as 'struct' here
  180 |   struct error_code
      |          ^~~~~~~~~~
/home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/system_error:55:9: note: replace the class-key with 'struct'
   55 |   class error_condition;
      |         ^~~~~~~~~~~~~~~
/home/ilg/.local/xPacks/@xpack-dev-tools/riscv-none-embed-gcc/10.1.0-1.1.1/.content/riscv-none-embed/include/c++/10.1.0/system_error:278:10: note: 'std::error_condition' defined as 'struct' here
  278 |   struct error_condition
      |          ^~~~~~~~~~~~~~~
```

The warning does not show when I compile the same code with arm-none-eabi-gcc 10.2, so I don't know if this issue is specific to GCC 10.1, or Arm patched their distribution.

**Comment by jim-wilson** [comment]
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96063

was fixed in 10.2 for all targets.  So we will get the fix when we upgrade to gcc-10.2.

**Comment by ilg-ul** [comment]
Thank you.
